src/rpcTest/scala/io/iohk/ethereum/rpcTest/RpcApiTests.scala

```diff
@@ -0,0 +1,1138 @@
+    val emiterResponse = service.ethSendTransaction(contractCall(firstAccount.address, receipt2.getContractAddress, writeContract(emitValue, emitEvent))).send()
+
+    val pendingTransactionLogs = service.ethGetFilterLogs(pendingTransactionFilterId).send()
+    pendingTransactionLogs.getLogs.asScala.size shouldEqual 2
+
+    val minedBlock1 = service.blockObservable(false).toBlocking.first()
+
+    // All filter which maps to test cases
+    // Used simple log filter that includes all txs, expected all txs logs to be shown
+    val filter = new EthFilter(getBlockParam(minedBlock1.getBlock.getNumber), latestBlock, List(firstAccount.address).asJava)
+
+    // Used log filter with higher fromBlock than toBlock, expected no change to be included
+    val badFilter =
+      new EthFilter(getBlockParam(minedBlock1.getBlock.getNumber.add(1)), getBlockParam(minedBlock1.getBlock.getNumber.subtract(1)), List(firstAccount.address).asJava)
+
+    // Used log filter that filters by event, expected only events selected to be shown
+    val particularEventTopic = new EthFilter(getBlockParam(minedBlock1.getBlock.getNumber), latestBlock, List(firstAccount.address).asJava)
+    particularEventTopic.addSingleTopic(counterContractEventHash)
+
+    // Used log filter with topic length larger than logs, expected no change to be included
+    val toLongTopic = new EthFilter(getBlockParam(minedBlock1.getBlock.getNumber), latestBlock, List(firstAccount.address).asJava)
+    toLongTopic
+      .addSingleTopic(counterContractEventHash)
+      .addSingleTopic(createTopic(firstAccount.address.drop(2)))
+      .addSingleTopic(createTopic(updateValue.toString))
+      .addSingleTopic(createTopic(updateValue.toString))
+
+    // Used log filter incluiding various contracts, expected changes from all contracts
+    val variousContracts =
+      new EthFilter(getBlockParam(minedBlock1.getBlock.getNumber), latestBlock, List(receipt.getContractAddress, receipt2.getContractAddress).asJava)
+
+    // Used log filter including only a null topic, expected changes to include all events
+    val nullTopic =
+      new EthFilter(getBlockParam(minedBlock1.getBlock.getNumber), latestBlock, List(receipt.getContractAddress, receipt2.getContractAddress).asJava)
+    nullTopic.addNullTopic()
+
+    // Install All filters
+    val installFilterId = service.ethNewFilter(filter).send().getFilterId
+    val badFilterId = service.ethNewFilter(badFilter).send().getFilterId
+    val particularEventTopicId = service.ethNewFilter(particularEventTopic).send().getFilterId
+    val toLongTopicId = service.ethNewFilter(toLongTopic).send().getFilterId
+    val variousContractsId = service.ethNewFilter(variousContracts).send().getFilterId
+    val nullTopicId = service.ethNewFilter(nullTopic).send().getFilterId
+
+    // BlockFilter shoul be empty
+    val blockFilterResponse = service.ethGetFilterLogs(blockFilterId).send()
+    blockFilterResponse.getLogs.asScala.size shouldEqual 0
+
+
```

Comment:
I think one enter here and in other places (lines: 115-116, 130-131, 307-308, 442-443, 484-489, 609-610, 622-623, 709-710) should work. This file is big itself maybe it could be divided into smaller parts, wdyt?

Reply:
I agree that file is rather large but for now i would like to leave it as it is to have all tests in one place. That way you can rather easily check if endpoint described in excel is tested in file.
Although in future, if we decide that we want more test cases ot automate this tests further then splitting this file may be first thing to do.